src/main/java/io/jmix/uisamples/bean/OverviewPageGenerator.java

```diff
@@ -0,0 +1,233 @@
+                }
+
+                flexLayout.add(verticalLayout);
+            }
+        }
+    }
+
+    private Component createTags(List<Element> tagElements) {
+        FlexLayout flexLayout = uiComponents.create(FlexLayout.class);
+        flexLayout.setAlignContent(FlexLayout.ContentAlignment.START);
+        flexLayout.setFlexDirection(FlexLayout.FlexDirection.ROW);
+        flexLayout.setJustifyContentMode(FlexComponent.JustifyContentMode.START);
+        flexLayout.setFlexWrap(FlexLayout.FlexWrap.WRAP);
+        flexLayout.addClassName(LumoUtility.Gap.SMALL);
+
+        for (Element tagElement : tagElements) {
+            Component tag = createComponentByName(tagElement.attributeValue("component"));
+
+            tag.getElement().getThemeList().add("badge pill");
+            addClassNames(tag, tagElement.attributeValue("classNames"));
+            if (tag instanceof HasText hasText) {
+                hasText.setText(tagElement.attributeValue("text"));
+            }
+            tag.addClassName(LumoUtility.FontWeight.LIGHT);
+
+            String route = tagElement.attributeValue("route");
+            flexLayout.add(StringUtils.isNotEmpty(route) ? createRoute(tag, route) : tag);
+        }
+
+        return flexLayout;
+    }
+
+    private Component createImage(Element imageElement) {
+        Image image = uiComponents.create(Image.class);
+        image.setSrc(imageElement.attributeValue("src"));
+        image.setWidth("20em");
+        String route = imageElement.attributeValue("route");
+        return StringUtils.isNotEmpty(route) ? createRoute(image, route) : image;
+    }
+
+    private Component createLabel(Element labelElement, String messagesPrefix) {
+        Component label = createLabel(getMessage(messagesPrefix,
+                labelElement.attributeValue("message")),
+                labelElement.attributeValue("classNames"),
+                labelElement.attributeValue("component"));
+        String route = labelElement.attributeValue("route");
+        return StringUtils.isNotEmpty(route) ? createRoute(label, route) : label;
+    }
+
+    private Component createRoute(Component component, String route) {
+        RouterLink routerLink = uiComponents.create(RouterLink.class);
+        RouteParameters routeParams = new RouteParameters("sampleId", route);
+        routerLink.setRoute(SampleView.class, routeParams);
+        routerLink.addClassNames("jmix-main-view-header-link");
+        routerLink.add(component);
+        return routerLink;
+    }
+
+    private Component createLabel(String text, String classNames, String componentName) {
+        Component component = createComponentByName(componentName);
+        if (component instanceof HasText hasText) {
+            hasText.setText(text);
+        }
+        addClassNames(component, classNames);
+
+        return component;
+    }
+
+    private Component createComponentByName(String componentName) {
+        if ("h1".equals(componentName)) {
```

Comment:
It will be more convenient to use switch expressions
```java
        return switch (componentName) {
            case "h1" -> uiComponents.create(H1.class);
            case "h2" -> uiComponents.create(H2.class);
            case "h3" -> uiComponents.create(H3.class);
            case "h4" -> uiComponents.create(H4.class);
            case "h5" -> uiComponents.create(H5.class);
            case "h6" -> uiComponents.create(H6.class);
            default -> uiComponents.create(Span.class);
        };
```